motr/m0kv/index_op.c

```diff
@@ -121,15 +144,31 @@ int index_drop(struct m0_realm *parent, struct m0_fid_arr *fids)
 	M0_PRE(fids != NULL && fids->af_count != 0);
 
 	for(i = 0; rc == 0 && i < fids->af_count; ++i) {
-		struct m0_idx   idx;
+		struct m0_idx   idx = {0};
 		struct m0_op   *op = NULL;
 
 		m0_fid_tassume(&fids->af_elems[i], &m0_dix_fid_type);
 		m0_idx_init(&idx, parent,
 			    (struct m0_uint128 *)&fids->af_elems[i]);
+
+		if (is_skip_layout) {
+			if (m0_fid_is_valid(&dix_pool_ver) && m0_fid_is_set(&dix_pool_ver))
+				set_enf_meta_flag(&idx);
+			else
+				return M0_ERR(-EINVAL);
+		} else if (m0_fid_is_set(&dix_pool_ver)) {
+			return M0_ERR(-EINVAL);
+		}
```

Comment:
This block is copied to 3 functions, can we avoid this copy-paste (may be move it to a function or something)?

Reply:
Done.